sympy/matrices/random.py

```diff
@@ -0,0 +1,2525 @@
+from random import Random
+
+from ..core import I as _i
+from ..core.mul import Mul as _multiply
+from ..functions import sqrt as _sqrt, re as _re, im as _im, \
+    transpose as _t, adjoint as _c
+from ..tensor import shape as _shape
+from .dense import eye as _eye
+from .inverse import _inv
+
+__all__ = 'projection', 'jordan', 'transposition', \
+          'permutation', 'elementary', 'rotation', 'reflection', \
+          'diagonal_normal', 'jordan_normal', 'isometry_normal', \
+          'triangular', 'invertible', 'singular', \
+          'idempotent', 'nilpotent', 'diagonalizable', 'trigonalizable', \
+          'orthogonal', 'unitary', 'normal', \
+          'symmetric', 'hermite', 'square', \
+          'regular_to_singular', 'complex_to_real', 'rand'
+
+
+_ALT = False
+_EPS = 1e-15
+
+
+# === random number generator functions ===
+
+
+rand = Random()
+r""" default random number generator
+
+    Explanation
+    ===========
+
+    the module default random generator
+    which is a instance of ``random.Random()``.
+    Hence, the state can be set by invoking ``rand.seed()``.
+
+
+    Examples
+    ========
+
+    >>> from sympy.matrices.random import rand, orthogonal
+
+    >>> rand.seed(1)
+    >>> orthogonal(3)
+    Matrix([
+    [-1,          0,          0],
+    [ 0,  sqrt(2)/2, -sqrt(2)/2],
+    [ 0, -sqrt(2)/2, -sqrt(2)/2]])
+
+    See Also
+    ========
+    projection
+
+"""
+
+
+def _rand(seed=None):
+    if hasattr(seed, 'sample'):
+        return seed
+    return rand if seed is None else Random(seed)
+
+
+def _sample(scalars, k=None, rng=None):
+    _k = k or 1
+    if hasattr(scalars, 'sample'):
+        smpl = scalars.sample(_k)
+    elif hasattr(rng, 'sample'):
+        smpl = rng.sample(scalars, _k)
+    else:
+        smpl = rand.sample(scalars, _k)
+    return smpl if k else smpl[0]
+
+
+# === helper ===
+
+
+def _set_value(o, i, j, v):
+    o[i, j] = v
```

Comment:
This function doesn't seem to be used any more

Reply:
yes, just forgot to remove it.